users/api/clients.go

```diff
@@ -35,7 +35,7 @@ func clientsHandler(svc users.Service, authClient auth.AuthClient, selfRegister
 	opts := []kithttp.ServerOption{
 		kithttp.ServerErrorEncoder(apiutil.LoggingErrorEncoder(logger, api.EncodeError)),
 	}
-	checkSuperAdminMiddleware := checkSuperAdminMiddleware(authClient)
+	checkSuperAdminMiddleware := api.CheckSuperAdminMiddleware(authClient)
 
 	r.Route("/users", func(r chi.Router) {
 		switch selfRegister {
```

Comment:
We can have self register in service layer, and with session we can detect the state authentication or login state.